File: Code/23_OpenCV/Ex09/Media_linhas_colunas.cpp
// Baseado em https://docs.opencv.org/master/db/df5/tutorial_linux_gcc_cmake.html

#include <stdio.h>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;

int main(int argc, char** argv)
{
	if(argc != 2)
	{
		printf("Modo de uso: %s Nome_Imagem\n",
			argv[0]);
		return -1;
	}
	Mat image = imread(argv[1]);
	Mat media_linhas, media_colunas;
	reduce(image, media_linhas, 0,
		CV_REDUCE_AVG, image.type());
	reduce(image, media_colunas, 1,
		CV_REDUCE_AVG, image.type());
	cout << " ==> Imagem: " << argv[1] << endl;
	cout << " Linhas: " << media_linhas << endl;
	cout << " Colunas: " << media_colunas << endl;
	return 0;
}

File: 4_Dicas/4.1_OpenCV/Ex09/Media_linhas_colunas.py
import cv2, numpy, sys

if len(sys.argv)<2:
	print("Modo de uso: " + sys.argv[0] + " Nome_Imagem")
	sys.exit(-1)
image = cv2.imread(sys.argv[1])
print(" ==> Imagem: " + sys.argv[1])
print(" Linhas: ")
print(cv2.reduce(image, 0, cv2.REDUCE_AVG))
print(" Colunas: ")
print(cv2.reduce(image, 1, cv2.REDUCE_AVG))